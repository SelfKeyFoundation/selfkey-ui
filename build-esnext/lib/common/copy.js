import * as React from 'react';
import { CopyToClipboard } from 'react-copy-to-clipboard';
import { withStyles, createStyles } from '@material-ui/core';
export const styles = createStyles({
    copyButton: {
        backgroundColor: 'transparent',
        border: '1px solid #3B4A5A',
        borderRadius: '4px',
        color: '#ffffff !important',
        cursor: 'pointer',
        display: 'flex',
        height: '24px',
        justifyContent: 'center',
        margin: '0 0 0 8px',
        minWidth: '63px',
        outline: 'none',
        padding: '8px 16px'
    },
    copyButtonSpan: {
        color: '#93b0c1',
        fontFamily: 'Lato, arial, sans-serif',
        fontSize: '10px',
        letterSpacing: 'normal',
    }
});
export class CopyComponent extends React.Component {
    constructor() {
        super(...arguments);
        this.copyText = 'COPY';
        this.copiedText = 'COPIED';
        this.state = {
            copyTextPlaceholder: this.copyText
        };
    }
    handleOnCopy() {
        return () => {
            this.setState({ copyTextPlaceholder: this.copiedText });
            const bounceTime = setTimeout(() => {
                this.setState({ copyTextPlaceholder: this.copyText });
                clearTimeout(bounceTime);
            }, 1000);
        };
    }
    render() {
        const { text, classes } = this.props;
        return (React.createElement(CopyToClipboard, { text: text, onCopy: this.handleOnCopy() },
            React.createElement("button", { className: classes.copyButton },
                React.createElement("span", { className: classes.copyButtonSpan }, this.state.copyTextPlaceholder))));
    }
}
export const Copy = withStyles(styles)(CopyComponent);
export default Copy;
//# sourceMappingURL=copy.js.map