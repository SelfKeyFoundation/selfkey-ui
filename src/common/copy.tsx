import * as React from "react";
import { CopyToClipboard } from "react-copy-to-clipboard";
// @ts-ignore
import injectSheet, { WithStyles, StyleSheet, StyledComponentProps } from 'react-jss';

export const styles: StyleSheet = {
  copyButton: {
    cursor: "pointer",
    width: "46px",
    height: "20px",
    marginLeft: "7px",
    padding: 0,
    backgroundColor: "#435160",
    border: "none",
    borderRadius: "4px",
    margin: 0,
    color: "#ffffff !important"
  },

  copyButtonSpan: {
    fontSize: "11px",
    lineHeight: "12px",
    fontFamily: 'Lato, arial, sans-serif',
    letterSpacing: "normal",
    color: "#93b0c1"
  }
};

export type CopyProps = {
  text: string;
};

export type CopyState = {
  copyTextPlaceholder: string;
};

export type StyledProps = WithStyles<keyof typeof styles> & CopyProps;


export class CopyComponent extends React.Component<StyledProps, CopyState> {
  copyText = "COPY";
  copiedText = "COPIED";

  state = {
    copyTextPlaceholder: this.copyText
  };

  constructor(props: StyledProps) {
    super(props);
  }

  handleOnCopy() {
    return () => {
      this.setState({ copyTextPlaceholder: this.copiedText });
      const bounceTime = setTimeout(() => {
        this.setState({ copyTextPlaceholder: this.copyText });
        clearTimeout(bounceTime);
      }, 1000);
    }
  }

  render() {
    const { text, classes } = this.props;
    return (
      <CopyToClipboard text={text} onCopy={this.handleOnCopy()}>
        <button className={classes.copyButton}>
          <span className={classes.copyButtonSpan}>
            {this.state.copyTextPlaceholder}
          </span>
        </button>
      </CopyToClipboard>
    );
  }
}

export const Copy = injectSheet(styles)(CopyComponent);

export default Copy;