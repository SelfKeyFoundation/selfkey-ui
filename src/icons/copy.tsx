import * as React from 'react';
import { WithStyles, withStyles, createStyles } from '@material-ui/core';
import { SvgIcon } from '@material-ui/core';
import { SvgIconProps } from '@material-ui/core/SvgIcon';

const styles = createStyles({
    root: {
        width: '16px !important',
        height: '16px !important',
    }
});

export const CopyIcon = withStyles(styles)((props: SvgIconProps & WithStyles<typeof styles>) => (
    <SvgIcon {...props} viewBox="0 0 14 16" version="1.1">
        <g id="ðŸ–¥-Main-Dashboard" fillRule="evenodd">
            <g id="302-Dashboard-Full-State-Expanded" transform="translate(-1125.000000, -2025.000000)">
                <g id="Transactions" transform="translate(150.000000, 1616.000000)">
                    <g id="Table" transform="translate(29.000000, 105.000000)">
                        <g id="Col4" transform="translate(946.000000, 4.000000)">
                            <g id="action" transform="translate(0.000000, 300.000000)">
                                <g id="copy">
                                    <path d="M13.5625,2.0625 C13.8541681,2.35416813 14,2.70833125 14,3.125 L14,11.5 C14,11.9166688 13.8541681,12.2708319 13.5625,12.5625 C13.2708319,12.8541681 12.9166687,13 12.5,13 L10,13 L10,14.5 C10,14.9166688 9.85416812,15.2708319 9.5625,15.5625 C9.27083188,15.8541681 8.91666875,16 8.5,16 L1.5,16 C1.08333125,16 0.729168125,15.8541681 0.4375,15.5625 C0.145831875,15.2708319 0,14.9166688 0,14.5 L0,4.5 C0,4.08333125 0.145831875,3.72916813 0.4375,3.4375 C0.729168125,3.14583187 1.08333125,3 1.5,3 L4,3 L4,1.5 C4,1.08333125 4.14583188,0.729168125 4.4375,0.4375 C4.72916812,0.145831875 5.08333125,0 5.5,0 L10.875,0 C11.2916687,0 11.6458319,0.145831875 11.9375,0.4375 L13.5625,2.0625 Z M8.3125,14.5 C8.35416688,14.5 8.39583312,14.4791669 8.4375,14.4375 C8.47916688,14.3958331 8.5,14.3541669 8.5,14.3125 L8.5,13 L5.5,13 C5.08333125,13 4.72916812,12.8541681 4.4375,12.5625 C4.14583188,12.2708319 4,11.9166688 4,11.5 L4,4.5 L1.6875,4.5 C1.64583312,4.5 1.60416688,4.52083313 1.5625,4.5625 C1.52083312,4.60416687 1.5,4.64583313 1.5,4.6875 L1.5,14.3125 C1.5,14.3541669 1.52083312,14.3958331 1.5625,14.4375 C1.60416688,14.4791669 1.64583312,14.5 1.6875,14.5 L8.3125,14.5 Z M12.3125,11.5 C12.3541669,11.5 12.3958331,11.4791669 12.4375,11.4375 C12.4791669,11.3958331 12.5,11.3541669 12.5,11.3125 L12.5,5 L9.75,5 C9.54166562,5 9.36458406,4.92708406 9.21875,4.78125 C9.07291594,4.63541594 9,4.45833438 9,4.25 L9,1.5 L5.6875,1.5 C5.64583312,1.5 5.60416688,1.52083313 5.5625,1.5625 C5.52083312,1.60416687 5.5,1.64583313 5.5,1.6875 L5.5,11.3125 C5.5,11.3541669 5.52083312,11.3958331 5.5625,11.4375 C5.60416688,11.4791669 5.64583312,11.5 5.6875,11.5 L12.3125,11.5 Z M12.5,3.5 L12.5,3.1875 C12.5,3.14583313 12.4791669,3.10416687 12.4375,3.0625 L10.9375,1.5625 C10.8958331,1.52083313 10.8541669,1.5 10.8125,1.5 L10.5,1.5 L10.5,3.5 L12.5,3.5 Z" id="icon-copy"></path>
                                </g>
                            </g>
                        </g>
                    </g>
                </g>
            </g>
        </g>
    </SvgIcon>
));

export default CopyIcon;
