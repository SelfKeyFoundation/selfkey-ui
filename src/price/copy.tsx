import React from 'react';
import { CopyToClipboard } from 'react-copy-to-clipboard';
import injectSheet from 'react-jss';
import { StyleSheet, StyledComponentProps } from '../react-jss.types';

const styles: StyleSheet = {
  copyButton: {
    cursor: 'pointer',
    width: '46px',
    height: '20px',
    marginLeft: '7px',
    padding: 0,
    backgroundColor: '#435160',
    border: 'none',
    borderRadius: '4px',
    margin: 0,
    color: '#ffffff !important'
  },

  copyButtonSpan: {
    fontSize: '11px',
    lineHeight: '12px',
    fontFamily: 'ProximaNovaRegular',
    letterSpacing: 'normal',
    color: '#93b0c1'
  },
};

interface Props {
  text: string;
}

export class CopyComponent extends React.Component<StyledComponentProps & Props, {}> {
  
  copyText = 'COPY';
  copiedText = 'COPIED';

  state = {
    copyTextPlaceholder: this.copyText
  };

  handleOnCopy() {
    this.state.copyTextPlaceholder = this.copiedText;
    const bounceTime = setTimeout( () => {
      this.state.copyTextPlaceholder = this.copyText;
      clearTimeout(bounceTime);
    }, 5000)
  };

  render() {
    return (
      <CopyToClipboard text={this.props.text}
          onCopy={this.handleOnCopy}>
        <button className={this.props.classes.copyButton}>
          <span className={this.props.classes.copyButtonSpan}>{this.state.copyTextPlaceholder}</span>
        </button>
      </CopyToClipboard>
    );
  }
}

export const Copy: React.Component<Props> = injectSheet(styles)(CopyComponent);

export default Copy;
