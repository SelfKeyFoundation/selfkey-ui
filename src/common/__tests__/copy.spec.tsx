import * as React from 'react';
import { create } from 'react-test-renderer';
import { Copy } from '../copy';

describe('Copy', () => {
	it('should Copy', () => {
		const component = create(<Copy text="Copy me!!" />);
		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
