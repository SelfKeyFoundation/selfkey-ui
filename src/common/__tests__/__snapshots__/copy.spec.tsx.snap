// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Copy should Copy 1`] = `
<button
  className="CopyComponent-copyButton-0-2-1"
  onClick={[Function]}
>
  <span
    className="CopyComponent-copyButtonSpan-0-2-2"
  >
    COPY
  </span>
</button>
`;
